src/lib/storage/table.cpp

```diff
@@ -492,31 +490,27 @@ size_t Table::memory_usage(const MemoryUsageCalculationMode mode) const {
   return bytes;
 }
 
-void Table::create_partial_hash_index(const ColumnID column_id, const std::vector<ChunkID>& chunk_ids,
-                                      const TableIndexType& type) {
-  std::shared_ptr<AbstractTableIndex> table_index = nullptr;
+void Table::_create_partial_hash_index(const ColumnID column_id, const std::vector<ChunkID>& chunk_ids) {
+  std::shared_ptr<PartialHashIndex> table_index = nullptr;
   std::vector<std::pair<ChunkID, std::shared_ptr<Chunk>>> chunks_to_index;
 
-  // If no ChunkIDs are passed an empty index is created.
   if (chunk_ids.empty()) {
-    const auto column_data_type = _column_definitions[column_id].data_type;
-    table_index = std::make_shared<PartialHashIndex>(column_data_type, column_id);
-  } else {
-    chunks_to_index.reserve(chunk_ids.size());
-
-    for (const auto& chunk_id : chunk_ids) {
-      const auto& chunk = get_chunk(chunk_id);
-      Assert(chunk, "Requested index on deleted chunk.");
-      Assert(!chunk->is_mutable(), "Cannot index mutable chunk.");
-      chunks_to_index.emplace_back(chunk_id, chunk);
-    }
+    Fail("Creating a partial hash index with no chunks being indexed is not supported.");
+  }
 
-    table_index = std::make_shared<PartialHashIndex>(chunks_to_index, column_id);
+  chunks_to_index.reserve(chunk_ids.size());
+  for (const auto& chunk_id : chunk_ids) {
+    const auto& chunk = get_chunk(chunk_id);
+    Assert(chunk, "Requested index on deleted chunk.");
+    Assert(!chunk->is_mutable(), "Cannot index mutable chunk.");
+    chunks_to_index.emplace_back(chunk_id, chunk);
   }
 
+  table_index = std::make_shared<PartialHashIndex>(chunks_to_index, column_id);
+
   _table_indexes.emplace_back(table_index);
```

Comment:
This would be fine since the idea of partial indexes allows having multiple indexes for the same column since each index could cover a different set of segments. Utilizing either one or multiple indexes is then an implementation detail of the caller. The index join, for example, assumes one index and uses the first index of the vector returned by `get_table_indexes` (see performance warning `There⋅are⋅multiple⋅table⋅indexes⋅available,⋅but⋅only⋅the⋅first⋅one⋅is⋅used.`).